src/components/Modals/centerModals/DatePicker/DatePicker.tsx

```diff
@@ -2,61 +2,56 @@ import { useState } from 'react';
 import { TouchableOpacity, View } from 'react-native';
 import DatePicker from 'react-native-modern-datepicker';
 import { useSelector } from 'react-redux';
+import Icon from 'src/components/Icon/Icon';
 import { getColors } from 'src/store/selectors';
-import { RED_200, WHITE_100 } from 'src/styles';
+import { WHITE_100 } from 'src/styles';
+import { IconName } from 'src/types';
+import { getFormattedDate, getInitialDate } from 'src/utils';
 
 import { styles } from './DatePicker.style';
-import { getFormattedDate, getInitialDate } from './DatePicker.utils';
 import Text from '../../../Text/Text';
 import ModalContainer from '../../ModalContainer/ModalContainer';
-import Icon from 'src/components/Icon/Icon';
-import { IconName } from 'src/types';
 
 type Props = {
-  error?: string;
   initialValue?: string;
   label?: string;
   minDate?: string;
   maxDate?: string;
   onDateChange: (selectedDate: string) => void;
 };
 
-const DateTimePicker = ({ error, initialValue, label, minDate, maxDate, onDateChange }: Props) => {
+const DateTimePicker = ({ initialValue, label, minDate, maxDate, onDateChange }: Props) => {
   const colors = useSelector(getColors);
-  const initialDate = getInitialDate({ initialValue });
+  const initialDate = getInitialDate(initialValue);
 
   const [showDatePicker, setShowDatePicker] = useState<boolean>(false);
-  const [selectedDate, setSelectedDate] = useState<string>(initialDate);
+  const [selectedDate, setSelectedDate] = useState<string | undefined>(initialDate);
```

Comment:
Why do you need a union with undefined type here?

Reply:
Because initialDate value sometimes come as undefined when editing accommodation or kinda related operations